builds/misc/templates/base-image-publish/base-image-setup.yaml

```diff
@@ -0,0 +1,50 @@
+steps:
```

Comment:
This is better, but you're still distributing the work across multiple templates so you're still doing all the serialization/deserialization. Templates are good for sharing code, but they're bad for organizing code because passing information between templates requires a lot of effort. And you're not sharing this logic with any other pipelines so there's not a compelling reason to use templates. If you want organization, you can use powershell functions inside a single script.

`base-image-build-and-publish.yaml`, for example, is only about 7 lines of code if you remove the deserialization and input checking logic. You could easily merge the powershell scripts from `base-image-setup.yaml`, `base-image-build-and-publish.yaml`, and `base-image-remove-librocksdb.yaml` into one script, and it would make things a lot easier to read and maintain.

And since

Reply:
😆 That how it was all started by one giant yaml script. Looks like this pipeline has come a full circle: dev -> create one giant yaml script -> thinking the script is too large -> overly refactored scripts -> decided the first version is the most readible one -> create one giant script.

Reply:
Haha...you definitely need organization and factoring, it's just that breaking up the powershell script into different yaml templates isn't ideal. If you need some organization you can break up the powershell code into functions instead, but still keep it all in one script.

Also, some amount of templates are fine since you don't want to duplicate the logic in each job, so that's why I suggested you just merge `base-image-setup.yaml`, `base-image-build-and-publish.yaml`, and `base-image-remove-librocksdb.yaml` into one yaml template with one powershell script.

Another option is to put the powershell into a ps1 file (which means you could also run it on dev desktop if needed), then call the ps1 script from each job. Either is fine with me.

Reply:
Alright, I merge the templates into a template since it's under 100... it should be ok? 
I'm not sure how I feel about having *.ps1 scripts with each executing only a sliver of a for-loop.